### PR TITLE
common: Mutex -> ceph::mutex

### PR DESCRIPTION
**Pull request by liewegas**
This is everything in common/ except for a few things that consume Mutex* from outsiders:

- SafeTimer
- Throttle
- obj_bencher

### CODE REVIEW

File: src/common/WorkQueue.cc [hunk 1]
```diff
@@ -302,36 +299,34 @@ void ShardedThreadPool::shardedthreadpool_worker(uint32_t thread_index)
...
       while (pause_threads) {
        cct->get_heartbeat_map()->reset_timeout(
 	        hb,
 	        wq->timeout_interval, wq->suicide_interval);
-       shardedpool_cond.WaitInterval(shardedpool_lock,
-	   utime_t(
-	   cct->_conf->threadpool_empty_queue_max_wait, 0));
+       shardedpool_cond.wait_for(
+	 ul,
+	 ceph::make_timespan(cct->_conf->threadpool_empty_queue_max_wait));
```
Comment (cbodley) [inline src/common/WorkQueue.cc]: make_timespan(...) involves conversion to/from double. i think std::chrono::seconds(...) would be clearer about the units

**Comment by liewegas** [comment]
fixing!

**Comment by liewegas** [comment]
@adamemerson will follow up shortly with something that removes the convenience.h helpers (and users)

**Comment by liewegas** [comment]
@tchaikov not sure what this failure is about: 
```
In file included from /home/jenkins-build/build/workspace/ceph-pull-requests/src/msg/Message.h:24:0,
                 from /home/jenkins-build/build/workspace/ceph-pull-requests/src/auth/none/AuthNoneSessionHandler.h:16,
                 from /home/jenkins-build/build/workspace/ceph-pull-requests/src/auth/AuthSessionHandler.cc:18:
/home/jenkins-build/build/workspace/ceph-pull-requests/src/common/RefCountedObj.h:87:9: error: 'condition_variable' in namespace 'ceph' does not name a type
   ceph::condition_variable cond;
         ^~~~~~~~~~~~~~~~~~
/home/jenkins-build/build/workspace/ceph-pull-requests/src/common/RefCountedObj.h: In member function 'int RefCountedCond::wait()':
/home/jenkins-build/build/workspace/ceph-pull-requests/src/common/RefCountedObj.h:95:7: error: 'cond' was not declared in this scope
       cond.wait(l);
       ^~~~
/home/jenkins-build/build/workspace/ceph-pull-requests/src/common/RefCountedObj.h:95:7: note: suggested alternative: 'Cond'
       cond.wait(l);
       ^~~~
       Cond
/home/jenkins-build/build/workspace/ceph-pull-requests/src/common/RefCountedObj.h: In member function 'void RefCountedCond::done(int)':
/home/jenkins-build/build/workspace/ceph-pull-requests/src/common/RefCountedObj.h:104:5: error: 'cond' was not declared in this scope
     cond.notify_all();
     ^~~~
/home/jenkins-build/build/workspace/ceph-pull-requests/src/common/RefCountedObj.h:104:5: note: suggested alternative: 'Cond'
     cond.notify_all();
     ^~~~
     Cond
src/auth/CMakeFiles/crimson-auth.dir/build.make:134: recipe for target 'src/auth/CMakeFiles/crimson-auth.dir/AuthSessionHandler.cc.o' failed
make[3]: *** [src/auth/CMakeFiles/crimson-auth.dir/AuthSessionHandler.cc.o] Error 1
make[3]: *** Waiting for unfinished jobs....
```

**Comment by liewegas** [comment]
(Also, I tried to cmake -DWITH_SEASTAR=On but I get all kinds of missing dependencies.. is that expected?)

**Comment by tchaikov** [comment]
@liewegas 
because with seastar, the interface of condition variable is very different than that in alien world, to avoid buggy implementation without being noticed, i didn't implement or add a workaround of `ceph::condition_variable` for `WITH_SEASTAR` like i did for mutex or shared_mutex.

instead, i'd suggest just disable these classes implemented using condition variable if `WITH_SEASTAR` is defined, like:

```diff
diff --git a/src/common/RefCountedObj.h b/src/common/RefCountedObj.h
index f96c399855..4bc79e891a 100644
--- a/src/common/RefCountedObj.h
+++ b/src/common/RefCountedObj.h
@@ -75,6 +75,7 @@ public:
   }
 };

+#ifndef WITH_SEASTAR
 /**
  * RefCountedCond
  *
@@ -163,6 +164,8 @@ struct RefCountedWaitObject {
   }
 };

+#endif // WITH_SEASTAR
+
 static inline void intrusive_ptr_add_ref(const RefCountedObject *p) {
   p->get();
 }
```

**Comment by liewegas** [comment]
retest this please

**Comment by tchaikov** [comment]
i am posting #25105 to rebase this PR onto master.

**Pull request closed without merging**